lib/convection/control/cloud.rb

```diff
@@ -16,17 +16,40 @@ def stacks
         @cloudfile.stacks
       end
 
-      def deck
-        @cloudfile.deck
+      def stack_groups
+        @cloudfile.stack_groups
       end
 
-      def converge(to_stack, &block)
-        unless to_stack.nil? || stacks.include?(to_stack)
-          block.call(Model::Event.new(:error, "Stack #{ to_stack } is not defined", :error)) if block
-          return
+      def filter_deck(options = {}, &block)
+        # throw an error if the user specifies both a stack group and list of stacks
+        if options[:stack_group] && options[:stacks]
+          block.call(Model::Event.new(:error, 'Cannot specify --stack-group and --stack-list at the same time', :error)) if block
+          return []
         end
 
-        deck.each do |stack|
+        # throw an error if the user specifies a nonexistent stack groups
+        if options[:stack_group] && !stack_groups.key?(options[:stack_group])
+          block.call(Model::Event.new(:error, "Unknown stack group: #{options[:stack_group]}", :error)) if block
+          return []
+        end
+
+        # throw an error if the user specifies nonexistent stacks
+        if Array(options[:stacks]).any? { |name| !@cloudfile.stacks.key?(name) }
+          bad_stack_names = options[:stacks].reject { |name| @cloudfile.stacks.key?(name) }
+          block.call(Model::Event.new(:error, "Stack#{'s' if bad_stack_names.length > 1}
+                    #{bad_stack_names.join(', ')} #{bad_stack_names.length > 1 ? 'do' : 'does'} not exist", :error)) if block
```

Comment:
Shall we create a private helper to generate this string? Truncating it like this is a bit confusing, at least for me, at first.